tests/test_issue_passed_parameters.py

```diff
@@ -0,0 +1,37 @@
+from unittest.mock import MagicMock
+
+from kink import di, inject
+
+def test_do_not_resolve_passed_kwargs() -> None:
+    di["name"] = "Bob"
+    @inject
+    class ExpensiveObject:
+        def __init__(self, name: str):
+            self.name = name
+            raise Exception("Constructing expensive object")
+
+    @inject
+    def greet(namer: ExpensiveObject, greeting: str = "Hello %s") -> str:
```

Comment:
@aikidojohn, the fact that this works in such scenarios is a side effect I allowed in the initial design, but it was not my intention for the library to be used in this way. I find it concerning when functions are called without parameters, despite their signature indicating otherwise.